### ISSUE TITLE
Add search.exclude option

### ISSUE THREAD

**Issue by CrossR** [onivim/oni2#2115]
As far as I can see, we are missing the `search.exclude` option, which is the same as `files.exclude` but specific to search.
It should also inherit all the values from `files.exclude`.

Technically this is what we should be using for the quick open as well according to the VSCode docs: 

```json
 // Configure glob patterns for excluding files and folders in fulltext searches and quick open. Inherits all glob patterns from the `files.exclude` setting.
  "search.exclude": {
    "**/node_modules": true,
    "**/bower_components": true,
    "**/*.code-search": true
  },
```

I'll tag this as a good first issue, since I don't think it should be too much work? We have the same functionality under a different name already, its just a case of adding this option, and also hooking it up to be used in `findInFiles`. https://github.com/onivim/oni2/pull/1099 hooked this up, but does not include the addition of the config option itself, and this option would need to apply to both `search` and `files.exclude`.

At a later date, we may want to update them both since right now we have a list of strings, rather than the key/value pairs that VSCode uses (not sure when that was changed).

**Comment by JonathanLorimer** [comment]
Hello, I would love to help out with this. I am new to OCaml / ReasonML, but I have a fair bit of experience with Haskell and Javascript / Typescript. If you could point me to the relevant files that I should start looking at I will try and get up to speed on this!

**Comment by CrossR** [comment]
Most of the files you'll be interested in are over in the `src/Core/` folder. You'll see the few related to the configuration, with `ConfigurationParser` being where the actual JSON option gets transformed. I'm not sure if there is another config option that uses the `{"option": bool, "option2": bool}` format, so it could be implemented as a list of strings to start most likely, which should be easier (and there are other examples). The other files will contain the default values and types etc.

`Ripgrep` in that same folder is where we do the actual calls into `ripgrep`, so you'd want to add `searchExclude` to `findInFiles`, and you can probably follow the linked PR in the OP to see how the `filesExclude` arg was hooked up, which is very similar to this. The only real difference is that you need to add the mentioned config option as well, and then concat `search.exclude` and `files.exclude` before sending.

Hopefully that makes sense, and there isn't something massively difficult hiding that I've not spotted from a quick poke around. The discord in the README can be a nice way to get some quick help (either specifically to Oni2 in our discord, or for OCaml/Reason in general in their respective discords).

**Comment by JonathanLorimer** [comment]
@CrossR Thanks for the quick response. I will start looking into this!

**Comment by bryphe** [comment]
Implemented now in #3044 - thank you @joseeMDS !
